sdk-base/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPlugin.kt

```diff
@@ -325,4 +326,18 @@ interface CameraAnimationsPlugin : MapPlugin {
    * @param animators Variable number of [ValueAnimator]'s
    */
   fun playAnimatorsSequentially(vararg animators: ValueAnimator)
+
+  /**
+   * Animations may need to run on a background thread. Use this to call [Animator.start] to
+   * ensure the animations are called on the animation thread.
+   *
+   * The thread for animations can be specified in the AndroidManifest.xml.
+   * ``` xml
+   *  <meta-data
+   *    android:name="com.mapbox.maps.BackgroundAnimatorThread"
+   *    android:value="true" />
+   * ```
+   */
+  @MapboxExperimental
+  fun post(runnable: Runnable): Boolean
```

Comment:
We need background thread as error-prone solution for AA + Xiaomi case only. There's no need to expose additional method on `sdk-base` module as `extension-androidauto` has dependency on `sdk` module which has api dependencies on all other plugins. This means we could expose this method on `CameraAnimationsPluginImpl` directly.
Main idea - this solution is overall a hacky solution for minority of devices in a specific use-case (AA) and we need to make this as less visible as possible.

Reply:
Are you suggesting that we build a Xiaomi device detector? And then behave differently with those animations?

> this solution is overall a hacky solution for minority

I actually fully agree. The goal right now, is to try and figure out how to make animations work at all. Because they don't work. 

Another solution could be, build a new animation framework for Android Auto - but that is not a small project.

Reply:
This is exposed so that the [NavigationCamera](https://github.com/mapbox/mapbox-navigation-android/blob/5ea19cd0b777b4cbe6bf63cf1409cc5ac6216ab4/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt#L418) can have access to it.

https://github.com/mapbox/mapbox-maps-android/pull/1605#issuecomment-1218166308

Are you suggesting Android Auto should build a new `NavigationCamera` so that it can run animations on a background thread? Android Auto uses the navigation-sdk, the navigation-sdk uses the maps-sdk.